### PR TITLE
Initial fixes for IE compat of cloneFragment

### PR DESCRIPTION
**Pull request by Dtphelan1**
#### Is this adding or improving a _feature_ or fixing a _bug_?

Fixing a bug

#### What's the new behavior?

Allows basic copy and pasting on IE11.

#### How does this change work?

Change is pretty self-explanatory. Avoids using functions not defined by IE11 or lower by a) converting a NodeList object to an Array, and b) by checking for browser compatibility before using setData functions with arguments that may throw errors.

#### Have you checked that...?

* [x] The new code matches the existing patterns and styles.
* [x] The tests pass with `yarn test`.
* [x] The linter passes with `yarn lint`. (Fix errors with `yarn prettier`.)
* [x] The relevant examples still work. (Run examples with `yarn watch`.)

#### Does this fix any issues or need any specific reviewers?

Fixes: #1941 
Reviewers: No one in particular

### CODE REVIEW

File: packages/slate-react/src/utils/clone-fragment.js [hunk 1]
```diff
@@ -88,7 +89,7 @@ function cloneFragment(event, value, fragment = value.fragment) {
 
   // For browsers supporting it, we set the clipboard registers manually,
   // since the result is more predictable.
-  if (event.clipboardData && event.clipboardData.setData) {
+  if (event.clipboardData && event.clipboardData.setData && !IS_IE) {
```
Comment (tobiasandersen) [inline packages/slate-react/src/utils/clone-fragment.js]: Does this mean IE has those methods, but they don't work?

Comment (Dtphelan1) [reply packages/slate-react/src/utils/clone-fragment.js]: It supports the method, but in a very restricted sense. Specifically, IE doesn't support arbitrary MIME types or even common ones like 'text/plain'; it only accepts the strings "Text" (which gets mapped to 'text/plain') and "Url" (mapped to 'text/url-list') as types for calls to `DataTransfer.setData()`. That's why I suggest we skip the following block if we are in IE.

**Comment by Dtphelan1** [comment]
I'm having trouble getting the linter to pass (n.b. I'm on a Windows machine). When I run `yarn lint` on my end I get a litany of errors across files I haven't touched; that said, the build seems to only fail on the code I've added, and it seems prettier is indicating the error is with line endings (which would make sense). Could someone with experience on this front offer some help?

File: packages/slate-react/src/utils/clone-fragment.js (hunk 1 above)
Comment (ianstormtaylor) [reply packages/slate-react/src/utils/clone-fragment.js]: Sounds like this one can't be polyfilled, so I'd be down to accept it alone. Although can you add a `COMPAT:` comment (check other parts of codebase to see one) to explain why. Thanks!

**Comment by Dtphelan1** [comment]
@ianstormtaylor Would you take a look now and let me know if there is anything else you'd like me to include?

**Comment by ianstormtaylor** [comment]
Thanks @Dtphelan1.

**Pull request merged**